projects/onto-search/src/lib/onto-search-facet/dropdown-multi-select-facet/dropdown-multi-select-facet.component.ts

```diff
@@ -0,0 +1,70 @@
+import {Component, Input, TemplateRef} from '@angular/core';
+import {OntoSearchFacetComponent} from '../onto-search-facet.component';
+import {FormControl} from '@angular/forms';
+import {ReplaySubject} from 'rxjs';
+import {SearchFacetType} from '../models/search-facet-type';
+import {SearchFacetModel} from '../models/search-facet-model';
+import {untilComponentDestroyed} from '@w11k/ngx-componentdestroyed';
+import {MatSelectChange} from '@angular/material/select';
+
+@Component({
+  selector: 'onto-dropdown-multi-select-facet',
+  templateUrl: './dropdown-multi-select-facet.component.html',
+  styleUrls: ['./dropdown-multi-select-facet.component.scss']
+})
+export class DropdownMultiSelectFacetComponent extends OntoSearchFacetComponent {
+  @Input()
+  public facetTemplate: TemplateRef<any>;
+
+  @Input()
+  public facetTitleTemplate: TemplateRef<any>;
+
+  @Input()
+  public type: SearchFacetType;
+
+  /** control for the selected from multi-selection */
+  public selectedFormControl: FormControl = new FormControl();
+
+  /** control for the MatSelect filter keyword multi-selection */
+  public filtredFormControl: FormControl = new FormControl();
+
+  /** list of filtered by search keyword for multi-selection */
+  public filtered: ReplaySubject<SearchFacetModel[]> = new ReplaySubject<SearchFacetModel[]>(1);
+
+  constructor() {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.filtered.next(this.data.facetGroupData.slice());
+    this.selectedFormControl.setValue(this.data.selected);
+
+    this.filtredFormControl.valueChanges
+        .pipe(untilComponentDestroyed(this))
+        .subscribe(() => {
+          this.filter();
+        });
+  }
+
+  private filter(): void {
+    if (!this.data.facetGroupData) {
+      return;
+    }
+
+    let search = this.filtredFormControl.value;
+    if (!search) {
+      this.filtered.next(this.data.facetGroupData.slice());
+      return;
+    } else {
+      search = search.toLowerCase();
+    }
+
+    this.filtered.next(
+        this.data.facetGroupData.filter((item) => item.label.toLowerCase().indexOf(search) > -1)
+    );
+  }
+
+  public onSelection($event: MatSelectChange): void {
```

Comment:
This emits on every selection of item. Would it not be better if it emits only once when the panel is closed?
With every emit comes an api request for data.